bus/pubsub/consumer.go

```diff
@@ -0,0 +1,136 @@
+package pubsub
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/pcelvng/task/bus/info"
+	"google.golang.org/api/option"
+)
+
+type Consumer struct {
+	// Client is a Google Pub/Sub client scoped to a single project.
+	// Clients should be reused rather than being created as needed.
+	// A Client may be shared by multiple goroutines. Consumer Client
+	client *pubsub.Client
+	sub    *pubsub.Subscription
+
+	// context for clean shutdown
+	ctx  context.Context
+	cncl context.CancelFunc
+
+	info info.Consumer
+}
+
+func (o *Option) NewConsumer() (c *Consumer, err error) {
+	opts := make([]option.ClientOption, 0)
+
+	if o.Host != "" && o.Host != "/" {
+		fmt.Println("setting PUBSUB_EMULATOR_HOST as", o.Host)
+		os.Setenv("PUBSUB_EMULATOR_HOST", o.Host)
+	}
+
+	if o.ProjectID != "" {
+		fmt.Println("setting PUBSUB_PROJECT_ID as", o.ProjectID)
+		os.Setenv("PUBSUB_PROJECT_ID", o.ProjectID)
+	}
+
+	if o.Connections > 1 {
+		opts = append(opts, option.WithGRPCConnectionPool(o.Connections))
+	}
```

Comment:
Honestly, you can probably just remove the connections options

Reply:
I can remove it as an option, but I think it needs to be set to at least 1, in the pipleline code I saw this comment 
"if option.WithGRPCConnectionPool is called with 0 value then the Topic calls hang indefinitely"